### PR TITLE
[docs] Improve cluster/docker docs

### LINKED ISSUE #3427: [docs] Cluster (and AWS/GCE) usage is unclear
### Describe the problem

Cluster/cloud usage is sort of hard to find. We should update the docs so that this is very easily accessible

### PR DESCRIPTION
**Pull request by richardliaw**
- Surfaces local cluster usage
 - Increases visability of these instructions
 - Removes some docker docs (that are really out of scope for Ray
 documentation IMO)

Closes #3427 .

### CODE REVIEW

File: doc/source/using-ray-on-a-large-cluster.rst [hunk 1]
```diff
@@ -1,309 +0,0 @@
-Manual Cluster Setup on a Large Cluster
```
Comment (richardliaw) [inline doc/source/using-ray-on-a-large-cluster.rst]: This should be subsumed by autoscaler

File: doc/source/using-ray-and-docker-on-a-cluster.md [hunk 2]
```diff
@@ -1,236 +0,0 @@
-# Using Ray and Docker on a Cluster (Experimental)
```
Comment (richardliaw) [inline doc/source/using-ray-and-docker-on-a-cluster.md]: This should be subsumed by autoscaler (and also has a lot of information that seems to be out of scope for this project)

File: doc/source/using-ray-on-a-large-cluster.rst (hunk 1 above)
Comment (robertnishihara) [reply doc/source/using-ray-on-a-large-cluster.rst]: If it is, that's awesome, but is it actually?

When using parallel-ssh, you are starting the cluster from the head node. When using the autoscaler, *I think* you are starting the cluster from your laptop which has to ssh to the head node. These are almost the same thing but not quite. I wonder if we're eliminating some use cases.

Are there other differences? Or is that the only one?

Comment (richardliaw) [reply doc/source/using-ray-on-a-large-cluster.rst]: > If it is, that's awesome, but is it actually?
I'm pretty sure they're effectively the same.

> When using parallel-ssh, you are starting the cluster from the head node.
The `worker_setup_commands/worker_ray_commands` are run from the head node (not from your laptop)

> with the autoscaler, you are starting the cluster from your laptop which has to ssh to the head node.
The head node is started from your laptop. However, the rest of the nodes are not and do not require you (on your laptop) to hold a bash session on the head node.

File: doc/source/autoscaling.rst [hunk 3]
```diff
@@ -50,6 +52,28 @@ SSH into the head node and then run Ray programs with ``ray.init(redis_address="
     # Teardown the cluster
     $ ray down ray/python/ray/autoscaler/gcp/example-full.yaml
 
+Quick start (Local)
```
Comment (ericl) [inline doc/source/autoscaling.rst]: "Local" might be misleading, maybe say "Local cluster of machines"?

**Review by ericl — APPROVED** [review]
Looks good to me. I guess the parallel ssh is available in the older doc versions, we could also leave them in at the bottom as (deprecated) for now.

File: doc/source/autoscaling.rst (hunk 3 above)
Comment (robertnishihara) [reply doc/source/autoscaling.rst]: How about "Local" -> "Private"? As opposed to a "Public" cloud.

Comment (robertnishihara) [reply doc/source/autoscaling.rst]: I mean this suggestion not just for this line, but everywhere.

**Comment by robertnishihara** [comment]
Anyway, I'm fine with either removing the parallel-ssh stuff or leaving it as deprecated.
